applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_strain_energy_response_function.cpp

```diff
@@ -99,9 +99,19 @@ namespace Kratos
               rResponseGradient.resize(rDerivativesMatrix.size1(), false);
         rResponseGradient.clear();
 
-        // There will be a mistake, if body forces are considered. Because the elements are responsible for the body forces!
-
-          KRATOS_CATCH("")
+        const double numerical_limit = std::numeric_limits<double>::epsilon();
+        Vector acc_1 = ZeroVector(3);
+        Vector acc_2 = ZeroVector(3);
+        if (rAdjointElem.GetProperties().Has( VOLUME_ACCELERATION ))
+            acc_1 = rAdjointElem.GetProperties()[VOLUME_ACCELERATION];
+          
+        if( rAdjointElem.GetGeometry()[0].SolutionStepsDataHas(VOLUME_ACCELERATION)) 
+            acc_2 = rAdjointElem.GetGeometry()[0].FastGetSolutionStepValue(VOLUME_ACCELERATION);
```

Comment:
We are not interested in the actual value, but just if the norm is non zero: 
You could say `acc_1 += rAdjointElem.GetGeometry()`
then you only need the acc_1 vector and also only need to calculate the norm once.

Reply:
done